### PR TITLE
Feat/payuk firewall rules

### PR DESCRIPTION
**Pull request by dibyadhar**
#### 📲 What

Adding custom firewall rules that is going to be implemented in a project but using a toggle variable to keep it off form the generic implementation.

There is a issue as discussed here that results in tight coupling of the policy and rule groups for now, which might get remediated in future release.

https://github.com/hashicorp/terraform-provider-aws/issues/28100
https://github.com/hashicorp/terraform-provider-aws/issues/34956

#### 🤔 Why

Add additional tls alert rule.

#### 🛠 How

By using dynamic block and custom toggle variable to configure TLS Alert rule. 

#### 👀 Evidence

In Customer PR (i.e in the root module PR) there is an attached build log and aws snapshot for the deployment.

image

#### 🕵️ How to test

By toggling the variable on in the root module when enabling firewall feature

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: aws/modules/infrastructure_modules/vpc/firewall.tf
Comment (russellseymour) [inline aws/modules/infrastructure_modules/vpc/firewall.tf:1]: You have two dynamic resources that have the same name, won't the latter one overwrite the former?

File: aws/modules/infrastructure_modules/vpc/firewall_rules.tf
Comment (russellseymour) [inline aws/modules/infrastructure_modules/vpc/firewall_rules.tf:1]: These seem very specific rules. Are they required by default on all Network Firewalls that are deployed?
How are extra rules added when the module is used?
Can you think about how to have a config that can be passed into the module and it iterates over it?

Comment (russellseymour) [inline aws/modules/infrastructure_modules/vpc/firewall_rules.tf:1]: The capacity has been changed. I appreciate that this was already hardcoded, but if we are going to need to be able to change it I would like it to be a variable with a sensible default value.

**Comment by russellseymour** [comment]
Why has the PR template been cut short?
It should have th What, How, Why and evidence that things are working when it is deployed.

File: aws/modules/infrastructure_modules/vpc/firewall_rules.tf
Comment (dibyadhar) [reply aws/modules/infrastructure_modules/vpc/firewall_rules.tf:1]: Generally there were two default rules already present ICMP and Egress Allow Domain List
The other Pay.UK Specific rules I toggled via create_custom_rule parameter 
Yeah definitely I can have a try just want to have this as stop gap solution to accommodate other sprint task and then come back.

Comment (dibyadhar) [reply aws/modules/infrastructure_modules/vpc/firewall_rules.tf:1]: Yeah I can update this to a variable

File: aws/modules/infrastructure_modules/vpc/firewall.tf
Comment (dibyadhar) [reply aws/modules/infrastructure_modules/vpc/firewall.tf:1]: I believe due to the difference in content it didn't overwrite, as when tested it created 4 rules distinctively

File: aws/modules/infrastructure_modules/vpc/firewall_rules.tf
Comment (dibyadhar) [reply aws/modules/infrastructure_modules/vpc/firewall_rules.tf:1]: Updated now

File: aws/modules/infrastructure_modules/vpc/firewall.tf [hunk 1]
```diff
@@ -83,6 +83,20 @@ resource "aws_networkfirewall_firewall_policy" "policy" {
       resource_arn = aws_networkfirewall_rule_group.icmp_alert_fw_rule_group.0.arn
     }
 
+    dynamic "stateful_rule_group_reference" {
+      for_each = var.create_custom_rule ? [0] : []
```
Comment (balpurewal) [inline aws/modules/infrastructure_modules/vpc/firewall.tf:87]: This variable should be called "create_tls_alert_rule"

**Review by balpurewal — APPROVED** [review]
This is fine for now. It would be good to provide a list of your own custom rules but looks like that isn't possible from what you have said. This PR is just allowing us to turn on/off a custom alert tls rule which is fine.

File: aws/modules/infrastructure_modules/vpc/firewall.tf
Comment (russellseymour) [reply aws/modules/infrastructure_modules/vpc/firewall.tf:1]: 👍

**Comment by russellseymour** [comment]
This is much better thanks, as now we have some control over what is happening.

**Review by russellseymour — APPROVED** [review]
OK happy with this now

**Pull request merged**